### PR TITLE
[WFLY-17774] Update SmallRye OpenTelemetry to a version with Issue #223 fixed

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-17774

Upgrade SmallRye OpenTelemetry to 2.3.1
(Also fixes sr-otel issue #224)

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/subresource/SubResourceTestCase.java [hunk 1]
```diff
@@ -35,11 +30,15 @@
...
 import org.jboss.shrinkwrap.api.asset.EmptyAsset;
-import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/subresource/SubResourceTestCase.java:40]: @jasondlee I suspect you have an IDE set to reorder imports according to a style that isn't the WF code style.

**Review by jamezp — APPROVED** [review]
This looks good to me with the exception of the import order that Brian pointed out.

**Comment by jasondlee** [comment]
> This looks good to me with the exception of the import order that Brian pointed out.

Shoot. I thought I had fixed those. I'll get those fixed for real as soon as I can.

Thanks.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/subresource/SubResourceTestCase.java (hunk 1 above)
Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/subresource/SubResourceTestCase.java:40]: I did, but I fixed that, but I guess those IDEA settings are per-project. I fixed these, and I guess I'll figure out how to make that default everywhere. :)

**Comment by bstansberry** [comment]
Thanks @jasondlee

**Pull request merged**